### PR TITLE
e2e: watch by prefix

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

File: e2e/etcdctlv3_test.go [hunk 1]
```diff
@@ -390,6 +399,9 @@ func ctlV3Watch(cx ctlCtx, key, value string) error {
 	} else {
 		cmdArgs = append(cmdArgs, watchCmd...)
 	}
+	if !cx.interactive && cx.prefix {
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: `if cx.prefix`

File: e2e/etcdctlv3_test.go [hunk 2]
```diff
@@ -390,6 +399,9 @@ func ctlV3Watch(cx ctlCtx, key, value string) error {
...
+		cmdArgs = append(cmdArgs, "--prefix")
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: `watchCmd = ...`

File: e2e/etcdctlv3_test.go [hunk 3]
```diff
@@ -397,16 +409,22 @@ func ctlV3Watch(cx ctlCtx, key, value string) error {
 	}
 
 	if cx.interactive {
-		if err = proc.Send(strings.Join(watchCmd, " ") + "\r"); err != nil {
+		line := strings.Join(watchCmd, " ")
+		if cx.prefix {
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: delete?

File: e2e/etcdctlv3_test.go [hunk 4]
```diff
@@ -379,35 +387,45 @@ func ctlV3Del(cx ctlCtx, key string, num int) error {
...
-		watchCmd = append(watchCmd, "--rev", strconv.Itoa(cx.watchRevision))
-	}
-
-	cmdArgs := ctlV3PrefixArgs(cx.epc, cx.dialTimeout)
+func ctlV3Watch(cx ctlCtx, key string, kvs ...kv) error {
+	watchCmd := append(ctlV3PrefixArgs(cx.epc, cx.dialTimeout), "watch")
 	if cx.interactive {
-		cmdArgs = append(cmdArgs, "watch", "--interactive")
+		watchCmd = append(watchCmd, "--interactive")
 	} else {
```
Comment (gyuho) [inline e2e/etcdctlv3_test.go]: @heyitsanthony https://github.com/coreos/etcd/blob/master/etcdctl/ctlv3/command/watch_command.go#L97-L102

watch command has its own parsing calls in interactive mode, so I had to generate the
command separately.

PTAL.

Thanks.

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
Thanks. Will merge after green lights.

**Pull request merged**